src/main/java/org/gradle/profiler/ScenarioLoader.java

```diff
@@ -298,6 +246,128 @@ private static Config getConfig(File scenarioFile, InvocationSettings settings,
         return executionInstructions;
     }
 
+    private static List<ScenarioDefinition> getScenarioDefinitions(File scenarioFile,
+        InvocationSettings settings, String scenarioName, Config scenario, String title,
+        List<BuildMutator> mutators, int buildCount, File scenarioBaseDir,
+        GradleBuildConfigurationReader inspector) {
+        List<? extends ScenarioDefinition> scenarioDefinitions;
+        if (scenario.hasPath(BAZEL) && settings.isBazel()) {
+            scenarioDefinitions = getBazelScenarioDefinitions(
+                scenarioFile, settings, scenarioName, scenario, title, mutators, buildCount,
+                scenarioBaseDir);
+        } else if (scenario.hasPath(BUCK) && settings.isBuck()) {
+            scenarioDefinitions = getBuckScenarioDefinitions(
+                scenarioFile, settings, scenarioName, scenario, title, mutators, buildCount,
+                scenarioBaseDir);
+        } else if (scenario.hasPath(MAVEN) && settings.isMaven()) {
+            scenarioDefinitions = getMavenScenarioDefinitions(
+                scenarioFile, settings, scenarioName, scenario, title, mutators, buildCount,
+                scenarioBaseDir);
+        } else if (!settings.isBazel() && !settings.isBuck() && !settings.isMaven()) {
+            scenarioDefinitions = getGradleScenarioDefinitions(
+                scenarioFile, settings, scenarioName, scenario, title, mutators, buildCount,
+                scenarioBaseDir, inspector
+            );
+        } else {
+            scenarioDefinitions = Collections.emptyList();
+        }
+        return Collections.unmodifiableList(scenarioDefinitions);
```

Comment:
Can we use an `ImmutableList.Builder` here? Another option would be to return early in each clause of the `if`, and make sure that the methods called would return `ImmutableList`'s.